### DISCUSSION TITLE
External plugin translations not compiled/shown

### DISCUSSION THREAD

**Discussion by juliusstoerrle** [pretix/pretix#3050] [Q&A]
Hi all,

We have a pretix installation running, but unfortunatly all of the plugins installed via pip do not get translated.
Our target is `de_Informal`, however it doesn't work with others either. All translations seem to be included in the plugin files.

During the `make production`, however these do not get compiled to .mo files. And all strings shown in the ticket shop are in English.

Sample dockerfile:
```
FROM pretix/standalone:stable
USER root
RUN pip3 install pretix-sepadebit
RUN pip3 install pretix-question-placeholders
RUN pip3 install pretix-pages
RUN pip3 install pretix-passbook
RUN pip3 install pretix-zugferd
RUN pip3 install pretix-covid-certificates
RUN pip3 install pretix-ldap
USER pretixuser
RUN cd /pretix/src && make production
VOLUME ["/etc/pretix", "/data"]
EXPOSE 80
ENTRYPOINT ["pretix"]
CMD ["all"]
```

I could not find anyone mentioning this issue anywhere. Our dockerfile seems to match the documentations one.
Does anyone experience the same or know a solution?

**Comment by raphaelm** [comment]
Interesting, we did not have heard this being reported before, so I'm not aware of any specific solution yet.

**Comment by juliusstoerrle** [comment]
This seems to be at least partially a permission issue.
When running msgfmt in the site-packages dir, it fails while using the pretix user.
Whenn adding the path to a plugins locale folder as custom translation path in the pretix.cfg, the make fails.
All fails relate to missing writing permissions.

If I add the following to the Dockerfile after plugin installs, before switching to the pretix user most plugins appear translated:

`RUN find /usr/local/lib/python3.9/site-packages/pretix_*/locale -name django.po -execdir msgfmt django.po -o django.mo \;`

Only `pretix-ldap` is not, but I think its simply not translated.

I am not that familiar with the translation process. Would it make sense to use venv's for plugin installation in a docker container? Would be great to have a more permanent solution instead of this workaround.
